demo/core_32_$.jssp.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Code City: Templating system for Code City.
+ *
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview JavaScript Server Pages for Code City.
+ * @author frase@google.com (Neil Fraser)
+ */
+
+$.jssp = {};
+
+$.jssp.compile = function(func) {
+  var src = func.jssp;
+  if (!src) {
+    throw ReferenceError('jssp not found on function.');
+  }
+  var tokens = src.trim().split(/(<%(?:--|=)?|(?:--)?%>)/);
```

Comment:
TIL you can capture bits of your separators and splice them into the split result array.  Nice!

But: why not capture the whole <%...%> as a single token?  That would simplify the main loop, because you wouldn't need to maintain any state—you could just examine the first few characters of each token.